internal/component/otelcol/internal/scheduler/scheduler.go

```diff
@@ -90,6 +115,10 @@ func (cs *Scheduler) Run(ctx context.Context) error {
 		case <-ctx.Done():
 			return nil
 		case <-cs.newComponentsCh:
+			if !firstRun { // do not pause on first run
+				cs.onPause()
+				firstRun = false
```

Comment:
firstRun cannot be set to false inside of the !firstRun statement. I think you should set is right after it, right?